File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/Archiwum/2022w/ID07IO1/Buider/src/IdBuilder.java

import java.util.HashSet;

public class IdBuilder
{
   private final HashSet<Long> data=new HashSet<>();
   /// .... where Id in (3,7,9)
   public void add(long id) { data.add(id); }
   public void remove(long id) { data.remove(id); }
   public void clear() { data.clear(); }

   @Override public String toString()
   {
      StringBuilder sb=new StringBuilder();
      boolean first=true;
      //data.stream().forEach((id)->sb.append(id));
      if(data.size()>0)
      {
         for(long id:data)
         {
            if(first) first=false; else sb.append(',');
            sb.append(id);
         }
      }
      else sb.append("null");
      return sb.toString();
   }

}

File: semestr-8/Jezyk C#/alexander/Csharp/Archiwum/2021s/Id06TC2/Builder/Builder/IdBuilder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Builder
{
    public class IdBuilder
    {
        // ... where id in (1, 13, 666)
        // ... where id in (null)
        // ib.Clear().Append(1).Append(13).Append(666).SetDelimetr(",").ToString();
        private string IfEmpty="null";
        private string Delimetr=", ";
        private HashSet<long> set=new HashSet<long>();
        public IdBuilder Append(long id)
        {
            set.Add(id);
            return this;
        }
        public IdBuilder Clear()
        {
            set.Clear();
            return this;
        }
        public IdBuilder SetDelimetr(string Delimetr)
        {
            this.Delimetr=Delimetr;
            return this;
        }
        public IdBuilder SetIfEmpty(string IfEmpty)
        {
            this.IfEmpty=IfEmpty;
            return this;
        }
        public override string ToString()
        {
            if(set.Count<=0) return IfEmpty;
            StringBuilder sb=new StringBuilder();
            bool first=true;
            foreach(long id in set)
            {
                if(first) first=false; else sb.Append(Delimetr);
                sb.Append(id);
            }
            return sb.ToString();
        }
    }
}
